src/pageScanner/checks/table-has-headers.js

```
@@ -0,0 +1,58 @@
+		if ( rows.length === 0 ) {
+			return true;
+		}
+
+		// Case 1: Valid row-header table (every row starts with <th scope="row">)
+		const isRowHeaderTable = rows.every( ( row ) => {
+			const firstCell = row.children[ 0 ];
+			if ( ! firstCell || firstCell.tagName.toLowerCase() !== 'th' ) {
+				return false;
+			}
+
+			return scope === 'row' || !scope;
```

ESLint

[error] 'scope' is not defined.

(no-undef)